artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ServerConsumerImpl.java

```diff
@@ -928,7 +927,7 @@ public Queue getQueue() {
 
             acks++;
          }
-         while (ref.getMessageID() != messageID);
+         while (ref != null && ref.getMessageID() != messageID);
```

Comment:
was this on purpose? it seems a left over from a debug, since it was never supposed to hit null here (the throw exception is done when ref==null)

Reply:
This was also not on purpose.